client/task_runner.go

```diff
@@ -997,6 +1019,16 @@ func (r *TaskRunner) run() {
 	}
 }
 
+// cleanup calls Driver.Cleanup when a task is stopping. Errors are logged.
+func (r *TaskRunner) cleanup() {
+	if drv, err := r.createDriver(); err != nil {
+		r.logger.Printf("[WARN] client: error creating driver to cleanup resources: %v", err)
+	} else {
+		ctx := driver.NewExecContext(r.taskDir, r.alloc.ID)
+		drv.Cleanup(ctx, r.createdResources)
```

Comment:
Why return an error at all if it's not retriable? The current approach I've taken in Cleanup is that it logs but doesn't return an error for acceptable failures (like a conflict deleting an image because its still in use) and unknown errors are returned which cause a retry. This seems like an easier way to offer soft vs retriable errors than wrapping in a special type and asserting in TaskRunner.